# SWALO
Scaffolding with assembly likelihood optimization

## Files
scaffolds_human.tar.gz contains scaffolding results of Human chromosome 14 data

scaffolds_rest.tar.gz contains scaffolding results of other datasets used in the paper

## Scripts
scaffolding_scripts.tar.gz contains scripts needed to generate the results used in the paper presenting SWALO. To run the scripts download the tarball and run
```
tar xzf scaffolding_scripts.tar.gz
cd scaffolding_scripts
```
## Installation

To generate the results first the following need to be installed first:

* [Bowtie](http://bowtie-bio.sourceforge.net/index.shtml) and [Bowtie 2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)
* [Prerequisites] (https://github.com/martinghunt/Scaffolder-evaluation) for running the analysis scripts used in the paper [http://www.genomebiology.com/2014/15/3/R42](http://www.genomebiology.com/2014/15/3/R42)
  * Some of the prerequisites can be installed by running `./installPrereqs`
* [MUMmer](https://sourceforge.net/projects/mummer/files/) for running evaluation scripts from the GAGE project 

Then install [SWALO](http://atifrahman.github.io/SWALO/) and the [analysis scripts](https://github.com/martinghunt/Scaffolder-evaluation) by running
```
./installSWALO
./installScripts
```
## Generating Results

The results in the paper can be generated by running
```
./runall
```

This will call all the required scripts and produce scatter plots showing the results in the same folder as the scripts. The scatter plots will be in *eps* files with the prefix *scatter*. Following files of interest will be in each lowest level directory:
* scaffolds.fa: The scaffolds file in fasta format
* check_scaffolds.log: The output generated by running analysis scripts on scaffolds.fa
* results.txt: *number of correct joins* *number of incorrect joins* made by SWALO

This will also download data and scripts from the GAGE project, run SWALO on contigs generated by various assemblers and run scripts to evaluate the scaffolds generated by SWALO as well as the scaffolding modules built into those assemblers. The evaluation results will be in *txt* files with the prefix *results* in the directory named *GAGE*.  


The following is a brief description of the scripts called by *runall*
* downloadData: Sets up the directory structure and downloads data
* mapReads: Runs bowtie2 to map reads to contigs
* mapReads_bowtie: Runs bowtie to map reads to contigs
* runSWALO: Runs SWALO to generate scaffolds
* runCombined: Combines results of multiple libraries
* runLong: Applies SWALO with long insert libraries on results with short insert libraries
* runCheck: Runs analysis scripts of analyze scaffolds generated by SWALO
* generatePlots: Generates scatter plots.
* downloadGAGEdata: Downloads data and scripts from the GAGE project
* runSWALO_GAGE: Runs SWALO on contigs generated by various assemblers
* downloadReferences_GAGE: Downloads reference genomes to evaluate results
* runCheck_GAGE: Runs evaluation scripts and outputs evaluation results in text files
